pygmt/clib/session.py

```diff
@@ -1680,3 +1681,56 @@ def extract_region(self):
         if status != 0:
             raise GMTCLibError("Failed to extract region from current figure.")
         return wesn
+
+    def read_virtualfile(self, vfname):
+        """
+        Read data from a virtual file.
+
+        Parameters
+        ----------
+        vfname : str
+            Name of the virtual file to read.
+
+        Returns
+        -------
+        Pointer to the data, which can be casted into GMT data types.
+        """
+        c_read_virtualfile = self.get_libgmt_func(
+            "GMT_Read_VirtualFile",
+            argtypes=[
+                ctp.c_void_p,
+                ctp.c_char_p,
+            ],
+            restype=ctp.c_void_p,
+        )
+        return c_read_virtualfile(self.session_pointer, vfname.encode())
+
+    @contextmanager
+    def virtualfile_to_gmtdataset(self):
+        """
+        Create a virtual file for writing a GMT_GRID object.
+
+        Yields
+        ------
+        vfile : str
+            Name of the virtual file.
+        """
+        family = "GMT_IS_DATASET"
+        geometry = "GMT_IS_PLP"
+        with self.open_virtual_file(family, geometry, "GMT_OUT", None) as vfile:
+            yield vfile
+
+    def gmtdataset_to_vectors(self, vfile):
+        data = ctp.cast(self.read_virtualfile(f"{vfile}"), ctp.POINTER(GMT_DATASET))
+        ds = data.contents
+
+        vectors = []
+        for itble in range(ds.n_tables):
+            dtbl = ds.table[itble].contents
+            for iseg in range(dtbl.n_segments):
+                dseg = dtbl.segment[iseg].contents
+                for icol in range(dseg.n_columns):
+                    vectors.append(
+                        np.ctypeslib.as_array(dseg.data[icol], shape=(dseg.n_rows,))
+                    )
+        return vectors
```

Comment:
Not sure what's the Pythonic way to store multiple tables or segments in NumPy arrays.
For example, if I have a GMT multi-segment file
```
> 
1 2
3 4
> 
5 6
7 8
```
Currently, the code incorrectly returns 
```
[[1, 3], [2, 4], [5, 6], [7, 8]]
```
At least it should return
```
[[1, 3, 5, 7], [2, 4, 6, 8]]
```
but it's still NOT a multi-segment "array".